src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx

```diff
@@ -52,22 +62,33 @@ function PhoneNumberPage() {
     const validate = useCallback(
         (values: FormOnyxValues<typeof ONYXKEYS.FORMS.PERSONAL_DETAILS_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.PERSONAL_DETAILS_FORM> => {
             const errors: FormInputErrors<typeof ONYXKEYS.FORMS.PERSONAL_DETAILS_FORM> = {};
-            if (!ValidationUtils.isRequiredFulfilled(values[INPUT_IDS.PHONE_NUMBER])) {
+            const phoneNumberValue = values[INPUT_IDS.PHONE_NUMBER];
+
+            if (!isRequiredFulfilled(phoneNumberValue)) {
                 errors[INPUT_IDS.PHONE_NUMBER] = translate('common.error.fieldRequired');
+                return errors;
             }
-            const phoneNumberWithCountryCode = LoginUtils.appendCountryCode(values[INPUT_IDS.PHONE_NUMBER]);
-            const parsedPhoneNumber = PhoneNumberUtils.parsePhoneNumber(values[INPUT_IDS.PHONE_NUMBER]);
-            if (!parsedPhoneNumber.possible || !Str.isValidE164Phone(phoneNumberWithCountryCode.slice(0))) {
-                errors[INPUT_IDS.PHONE_NUMBER] = translate('bankAccount.error.phoneNumber');
+
+            if (!CONST.ACCEPTED_PHONE_CHARACTER_REGEX.test(phoneNumberValue) || CONST.REPEATED_SPECIAL_CHAR_PATTERN.test(phoneNumberValue)) {
+                errors[INPUT_IDS.PHONE_NUMBER] = translate('common.error.phoneNumber');
+                return errors;
             }
 
-            // Clear the error when the user tries to validate the form and there are errors
-            if (validateLoginError && !!errors) {
-                PersonalDetails.clearPhoneNumberError();
+            const sanitizedPhoneNumber = sanitizePhoneNumber(phoneNumberValue);
+            const phoneNumberWithCountryCode = appendCountryCode(sanitizedPhoneNumber);
+            const parsedPhoneNumber = formatPhoneNumber(phoneNumberValue);
+
+            if (!parsedPhoneNumber.possible || !Str.isValidE164Phone(phoneNumberWithCountryCode)) {
```

Comment:
```suggestion
            const parsedPhoneNumber = formatPhoneNumber(phoneNumberValue);
            if (!parsedPhoneNumber.possible || !Str.isValidE164Phone(parsedPhoneNumber.number.e164)) {
```

<img width="279" alt="Screenshot 2025-02-16 at 13 42 23" src="https://github.com/user-attachments/assets/4fb98f88-1f66-4752-bb36-c78a57f7e8f8" />


The function `formatPhoneNumber` already sanitizes the phone number, so we only need to validate E.164 from `parsedPhoneNumber` result

Or we can move this logic into a new function at `PhoneNumberUtil.isValidPhoneNumber` for better reusability.